### PR TITLE
align record fields with spaces after commas

### PR DESCRIPTION
**Pull request by cblp**

### CODE REVIEW

**Comment by chrisdone** [comment]
I think multiple lines is a bit aggressive for just a few fields. Can this detect whether it fits on one line first?

**Comment by mrBliss** [comment]
You have also removed the space between the constructor and the `{`. Either do this in a separate pull request or don't do it all (I'd like to believe I'm not alone in appreciating this extra space :wink:).

**Comment by cblp** [comment]
@chrisdone @mrBliss done

File: TESTS.md [hunk 1]
```diff
@@ -292,6 +292,28 @@ test
   ,)
 ```
 
+# Record syntax
+
+Pattern matching, short
+
+```haskell
+fun Rec {alpha = beta, gamma = delta, epsilon = zeta, eta = theta, iota = kappa} = do
+  beta + delta + zeta + theta + kappa
```
Comment (mrBliss) [inline TESTS.md]: Sorry to nitpick, but why is there no space *after* the `{` either (and before the `}`)? This is inconsistent with multi-line formatting.

Comment (cblp) [inline TESTS.md]: It is consistent with inline lists and tuples.

Comment (mrBliss) [inline TESTS.md]: You have a point. Unfortunately this is not mentioned in Johan Tibell's style guide.
What follows is my personal opinion on spacing around braces (`{}`).

What about one-line record declarations?

I prefer
```haskell
data User = User { getName :: String }
newtype Bourne a = Bourne { getIdentity :: a }
```
to
```haskell
data User = User {getName :: String}
newtype Bourne a = Bourne {getIdentity :: a}
```
To me it makes sense to consequently also prefer `Rec { foo = 1 }` to `Rec {foo = 1}`.

What about braces around blocks? `case x of { _ -> 1 }` vs `case x of {_ -> 1}` or `do { x <- run; return (x + 1) }` vs `do {x <- run; return (x + 1)}`.

Can't we special case braces (`{}`)?

Comment (cblp) [inline TESTS.md]: I think code block braces are very different syntactic case. I'd prefer to exclude them from this discussion.

As for record declarations:
1. (My personal opinion) Record declaration must be consistent with record access/update/etc syntax, and latter must be consistent with list and tuple syntax.
2. (My personal opinion) Record declaration must be consistent with type tuple.
```haskell
data User = User (Id, String)
newtype Bourne a = Bourne (Identity, a)
```

I'd love to hear @chrisdone's and @tibbe's thoughts.

**Comment by chrisdone** [comment]
I think this is good. The spaces come out on multiple-lines because of the commas.

(I prefer no spaces at all like Show's output, but we're adhering to a style.)

**Pull request merged**